sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_deserialize.py

```diff
@@ -153,8 +146,10 @@ def process_storage_error(storage_error):   # pylint:disable=too-many-statements
         pass
 
     try:
+        if error_code in [StorageErrorCode.blob_not_found]:
```

Comment:
Sorry should have mentioned this when we were talking about it, but can you move these outside the `try` and add a comment that says something like `Convert blob errors to datalake errors`. This is sort of obvious but I want it to be clear this is here so it can be found later.

Reply:
Good point- I will move it out and add a helpful comment for clarity!